meta-mbl-distro/scripts/create-update-payload

```diff
@@ -74,7 +67,48 @@ import argparse_oe
 
 
 MBL_BOOT_ID = "BOOT"
-PAYLOAD_FORMAT_VERSION = b"1"
+MBL_APPS_ID = "APPS"
+MBL_ROOTFS_ID = "ROOTFS"
+
+PAYLOAD_FORMAT_VERSION = b"2"
+
+verbose_file = None
+
+
+class ArchivedFileSpec:
+    """
+    A specification of a file to be added to an archive.
+
+    The class holds two pieces of information: the path to the file to be
+    added, and the path/filename that the file should have within the archive.
+    """
+
+    def __init__(self, path, archived_path=None):
+        """
+        Create a new ArchivedFileSpec.
+
+        :param path str|Path: path to the file to be archived. The path is
+            resolved so that if it points to a symlink, the underlying file
+            will be used rather than the symlink itself.
+        :param archived_path str|Path: path to the file within the archive. If
+            set to None (the default value) then the file will be added to the
+            archive with the same name as outside the archive (before symlink
+            resolution).
+        """
+        self.path = pathlib.Path(path)
+        if archived_path is None:
+            self.archived_path = path.name
+        else:
+            self.archived_path = pathlib.Path(archived_path)
+
+        # We don't want symlinks in our payloads
+        self.path = self.path.resolve()
+
+
+def print_verbose(s):
+    """Print only if in verbose mode"""
```

Comment:
Yeah, I considered that and then chose not to use it because we're already using BitBake's functions for loging fatal errors.

I wouldn't mind getting rid of the BitBake style error reporting just using a logger + exit codes or uncaught exceptions.